dsn.go

```diff
@@ -207,7 +207,8 @@ func ParseDSN(dsn string) (*Config, error) {
 	}
 
 	if parsed.Path != "" {
-		conf.schema = strings.TrimPrefix(parsed.Path, "/")
+		s := strings.Split(parsed.Path, "/")
+		conf.schema = s[len(s)-1]
```

Comment:
This is pretty subtle now, in that the trailing slash is important.

`https://some.host/proxied/avatica` is unique to `https://some.host/proxied/avatica/` in that the first is parsed as: `{'endpoint':'https://some.host/proxied', 'schema':'avatica'}` and the second is parsed as `{'endpoint':'https://some.host/proxied/avatica', 'schema':'avatica'}`.

Looking at other DSN's, this is a pretty common convention. However, most databases don't allow/expose querying from behind a normal http loadbalancer ;)

Is it going to be less error prone if we let the default schema to be provided via a query string instead of the path?

Reply:
Chatting with Josiah (coworker), we're both looking at what go-sql-driver/mysql does, and found https://github.com/go-sql-driver/mysql/blob/master/dsn.go#L168 / https://godoc.org/github.com/go-sql-driver/mysql#Config.FormatDSN

I think the "concerns" I have with the above could be done away with a similar method that can accept a struct and format the DSN for the user (to pass into `sql.Open()`)? WDYT @F21, you're the Go expert compared to me ;)

Reply:
@joshelser Good catch! Yes, that is pretty subtle and could be difficult to debug. I think there should not be a trailing slash after the schema (if provided) as this is the convention used by a few drivers I checked. This is the case for the MySQL driver you've linked as well as the postgres driver: https://pkg.go.dev/github.com/lib/pq?tab=doc

In other words, `https://some.host/proxied/avatica` means the default schema is `avatica` and `https://some.host/proxied/avatica/` means no default schema. As you mentioned, this is definitely a subtle difference, but at least we are following convention, so there's consistency across drivers.

I am not opposed to moving the default schema to a query string, but it seems to be a break from the standard convention and I don't think it's really necessary. In addition, that would be a breaking change as well, so if we do go down this path, we'd have to consider if we're breaking anything else.

I think the `FormatDSN()` method is a good idea. We can change the properties in the [`Config`](https://godoc.org/github.com/apache/calcite-avatica-go#Config) struct to public (breaking change, but additive) and add a `FormatDSN` method to turn the config back to the DSN.

The above probably sounds a bit rambly, but I personally prefer the following approach:
- `https://some.host/proxied/avatica` means the schema is `avatica` and `https://some.host/proxied/avatica/` means no schema.
- Let's make the properties in `Config` public and add a `FormatDSN()` method to it.

Reply:
Agree that while there is some subtlety it is consistent.

I’m happy to take a shot at FormatDSN().

Since these are separate - though related - things, I’ll probably start a
new ticket and branch for that effort.

Agree @F21? Are you happy with the solution and tests in this PR?

Reply:
@josiahg Can you add another test to test the case where there is a trailing slash and there's no default schema when the DSN is parsed? This is just so we don't break anything in the future if changes are made to the code that parses the DSN.

Other than that, it looks great! 👍

Reply:
> The above probably sounds a bit rambly, but I personally prefer the following approach:

I am OK with this! We can document this in the site and give some nice logging to help catch when a user runs into it.